src/rust/rust-proto-new/src/graplinc/grapl/api/pipeline_ingress/v1beta1.rs

```diff
@@ -0,0 +1,143 @@
+use bytes::{
+    Bytes,
+    BytesMut,
+};
+use prost::Message;
+
+use crate::{
+    graplinc::common::v1beta1::{
+        SystemTime,
+        Uuid,
+    },
+    protobufs::graplinc::grapl::api::pipeline_ingress::v1beta1::{
+        PublishRawLogsRequest as PublishRawLogsRequestProto,
+        PublishRawLogsResponse as PublishRawLogsResponseProto,
+    },
+    type_url,
+    SerDe,
+    SerDeError,
+};
+
+//
+// PublishRawLogsRequest
+//
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct PublishRawLogsRequest {
+    pub event_source_id: Uuid,
+    pub tenant_id: Uuid,
+    pub log_event: Bytes,
+}
+
+impl TryFrom<PublishRawLogsRequestProto> for PublishRawLogsRequest {
+    type Error = SerDeError;
+
+    fn try_from(
+        publish_raw_logs_request_proto: PublishRawLogsRequestProto,
+    ) -> Result<Self, Self::Error> {
+        let event_source_id = publish_raw_logs_request_proto
+            .event_source_id
+            .ok_or_else(|| SerDeError::MissingField("event_source_id".to_string()));
+
+        let tenant_id = publish_raw_logs_request_proto
+            .tenant_id
+            .ok_or_else(|| SerDeError::MissingField("tenant_id".to_string()));
+
+        Ok(PublishRawLogsRequest {
+            event_source_id: event_source_id?.into(),
+            tenant_id: tenant_id?.into(),
```

Comment:
Probably laziness.. `rust-analyzer` likely told me I had the wrong type at that site so I wrote a question mark because my cursor was already there. It would make a lot more sense at the place where it's declared, will fix.

Reply:
fixed